src/stores/wordpress-versions-api.ts

```diff
@@ -0,0 +1,167 @@
+
+const wordPressApiResponseSchema = z.object( {
+	offers: z.array( z.any() ),
+} );
+
+type WordPressApiOffer = {
+	version: string;
+	response?: string;
+	[ key: string ]: unknown;
+};
+
+type ProcessedOffer = {
+	version: string;
+	shortName: string;
+};
+
+const extractShortName = ( version: string ): string => {
+	if ( isWordPressDevVersion( version ) ) {
+		return 'nightly';
+	}
+	const match = version.match( /^(\d+\.\d+)/ );
+	return match ? match[ 1 ] : version;
+};
+
+export interface WordPressVersion {
+	isBeta: boolean;
+	isDevelopment: boolean;
+	isLatest: boolean;
+	label: string;
+	value: string;
+}
+
+async function fetchWordPressApiData( channel: 'beta' | 'development', version?: string ) {
+	const baseUrl = 'https://api.wordpress.org/core/version-check/1.7/';
+	const params = new URLSearchParams( { channel } );
+
+	if ( channel === 'beta' && version ) {
+		params.append( 'version', version );
+	}
+
+	const response = await fetch( `${ baseUrl }?${ params }` );
+	if ( ! response.ok ) {
+		throw new Error( 'Failed to fetch WordPress versions' );
+	}
+	return wordPressApiResponseSchema.parse( await response.json() );
+}
+
+function processWordPressOffers(
+	offers: WordPressApiOffer[],
+	isDevelopment = false,
+	shortNameOccurrences: Map< string, number >
+): ProcessedOffer[] {
+	return offers
+		.map( ( offer ) => wordPressOfferSchema.safeParse( offer ) )
+		.filter( ( result ): result is { success: true; data: WordPressOffer } => result.success )
+		.filter( ( result ) =>
+			isDevelopment ? result.data.response === 'development' : result.data.response === 'autoupdate'
+		)
+		.map( ( { data } ) => {
+			const shortName = extractShortName( data.version );
+			shortNameOccurrences.set( shortName, ( shortNameOccurrences.get( shortName ) || 0 ) + 1 );
+			return {
+				version: data.version,
+				shortName,
+			};
+		} );
+}
+
+function generateVersionLabel(
+	version: string,
+	shortName: string,
+	occurrences: number,
+	isLatest: boolean
+): string {
+	if ( isWordPressDevVersion( version ) ) {
+		return 'nightly';
+	}
+	if ( isLatest ) {
+		return sprintf( __( '%s (latest)' ), version );
+	}
+	if ( occurrences > 1 || isWordPressBetaVersion( version ) ) {
+		return version;
+	}
+	return shortName;
+}
+
+function findLatestStable( versions: ProcessedOffer[] ): ProcessedOffer | undefined {
+	return versions.find(
+		( version: ProcessedOffer ) =>
+			! isWordPressBetaVersion( version.version ) && ! isWordPressDevVersion( version.version )
+	);
+}
+
+export const wordpressVersionsApi = createApi( {
+	reducerPath: 'wordpressVersionsApi',
+	baseQuery: fetchBaseQuery(),
+	endpoints: ( builder ) => ( {
+		getWordPressVersions: builder.query< WordPressVersion[], void >( {
+			queryFn: async () => {
+				try {
+					const [ stableData, developmentData ] = await Promise.all( [
+						fetchWordPressApiData( 'beta', MINIMUM_WORDPRESS_VERSION ),
+						fetchWordPressApiData( 'development' ),
+					] );
```

Comment:
```suggestion
				try {
					// eslint-disable-next-line no-var
					var [ stableData, developmentData ] = await Promise.all( [
						fetchWordPressApiData( 'beta', MINIMUM_WORDPRESS_VERSION ),
						fetchWordPressApiData( 'development' ),
					] );
				} catch ( error ) {
					if ( error instanceof z.ZodError ) {
						Sentry.captureException( error );
					}
					throw error;
				}
```
Nit, but it'd be nice to reduce the indentation level in this function by only wrapping the fetch code in `try catch`. Doesn't have to be exactly like my suggestion, but something like this.

Reply:
Done! https://github.com/Automattic/studio/pull/1102/commits/66384622791bdb6f151f025d1029b4ab0a952f41
`processWordPressOffers` is using zod `safeParse` so I didn't include it into the try/catch.